test/pkg/environment/expectations.go

```diff
@@ -0,0 +1,69 @@
+package environment
+
+import (
+	"sync"
+
+	. "github.com/onsi/gomega" //nolint:revive,stylecheck
+	"github.com/samber/lo"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	storagev1 "k8s.io/api/storage/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/karpenter/pkg/apis/awsnodetemplate/v1alpha1"
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+)
+
+var (
+	EnvironmentLabelName = "testing.karpenter.sh/environment-name"
+	CleanableObjects     = []client.Object{
```

Comment:
How do we ensure that Nodes are cleaned up? I think the TTL on the Provisioners is meant to do that, but if a test fails to delete nodes (which could happen), then we want our Cleanup to also include those.

Reply:
Ah I forgot that Nodes are owned by provisioners now, so this should be good.

Reply:
We definitely need to test scale down, too. I think this library will need to be greatly expanded upon (e.g. wait for things to cleanup). This is just to get us started.